README.md

```diff
@@ -249,6 +249,33 @@ struct my_struct
 auto x = my_struct{$"id"|17, $"value"|42};
 ```
 
+## Create mock objects
+In [GMock](https://github.com/google/gmock) and [Trompe l'Oeil](https://github.com/rollbear/trompeloeil) macros are used manually to create mock objects. It is possible to do this automatically if you are willing to accept undefined behaviour, as shown with [FakeIt](https://github.com/eranpeer/fakeit) and [HippoMocks](https://github.com/dascandy/hippomocks). Combining this proposal with the reflexpr proposal extended with function reflection allows a mocking framework that does not invoke manual labor or undefined behaviour:
+
+```C++
+#include <experimental/reflexpr>
+
+using meta::get_base_name_v;
+using meta::get_return_type_t;
+using meta::get_functions_v;
+
+template <typename Interface, typename Function, size_t argCount>
+struct ReifyFunction {
+  virtual Interface* GetMock() = 0;
+  get_return_type_t<Function> $get_base_name_v<Function>() {
+    return MockingLibrary::HandleMockCall(GetMock(), get_base_name_v<Function>());
+  }
+};
+
+template <typename Interface>
+struct Mock : Interface, ReifyFunction<meta::get_functions_v<reflexpr(Interface)> >... {
```

Comment:
Understood. So the `...` unpacks the parameter pack delivered by `meta::get_functions_v`.

If you add a comment for that, I am happy :-)